data/lib/src/datasource_impl/document_datasource_impl.dart

```diff
@@ -97,4 +99,28 @@ class DocumentDataSourceImpl implements DocumentDataSource {
       });
     });
   }
+
+  @override
+  Future<DownloadTaskId> downloadDocument(DocumentId documentId, Token token, Uri baseUrl) async {
+    var externalStorageDirPath;
+    if (Platform.isAndroid) {
+      externalStorageDirPath = (await getExternalStorageDirectory()).absolute.path;
+    } else if (Platform.isIOS) {
+      // iOS Implementation
+    } else {
+      throw DeviceNotSupportedException();
+    }
+
+    final taskId = await FlutterDownloader.enqueue(
+      url: baseUrl.withDownloadPath(Endpoint.documents, documentId),
```

Comment:
refer to https://github.com/linagora/linshare-mobile-flutter-app/pull/13/files#r525022907

How about
```
baseUrl.withServicePath(Endpoint.documents).download(documentId);
```

Reply:
(may be it is not true, but that idea